File: xiang.li/1313.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int length = 0;
        for(int i = 0;i < nums.length;i += 2){
            length += nums[i];
        }

        int[] result = new int[length];
        int k = 0;

        for(int i = 0;i < nums.length;i += 2){
            while(nums[i] > 0) {
                result[k] = nums[i + 1];
                nums[i]--;
                k++;
            }
        }
        
        return result;
    }
}

File: hang.liu/1313.js
var decompressRLElist = function(nums) {
    var result = []
    for (var i = 0; i < nums.length; i += 2) {
        for (var j = 0; j < nums[i]; j++) {
            result.push(nums[i + 1])
        }
    }
    return result
};